project/MimaFilters.scala

```diff
@@ -66,13 +66,21 @@ object MimaFilters extends AutoPlugin {
     ProblemFilters.exclude[MissingClassProblem]("scala.reflect.io.FileZipArchive$zipFilePool$"),
 
     // #9727
-    ProblemFilters.exclude[DirectMissingMethodProblem]("scala.collection.concurrent.TrieMap.filterInPlaceImpl"),
-    ProblemFilters.exclude[DirectMissingMethodProblem]("scala.collection.concurrent.TrieMap.mapValuesInPlaceImpl"),
-    ProblemFilters.exclude[DirectMissingMethodProblem]("scala.collection.convert.JavaCollectionWrappers#JConcurrentMapWrapper.filterInPlaceImpl"),
-    ProblemFilters.exclude[DirectMissingMethodProblem]("scala.collection.convert.JavaCollectionWrappers#JConcurrentMapWrapper.mapValuesInPlaceImpl"),
+    ProblemFilters.exclude[DirectMissingMethodProblem]("scala.collection.concurrent.TrieMap.filterInPlaceImpl"),                                      // private[collection]
+    ProblemFilters.exclude[DirectMissingMethodProblem]("scala.collection.concurrent.TrieMap.mapValuesInPlaceImpl"),                                   // private[collection]
+    ProblemFilters.exclude[DirectMissingMethodProblem]("scala.collection.convert.JavaCollectionWrappers#JConcurrentMapWrapper.filterInPlaceImpl"),    // private[collection]
+    ProblemFilters.exclude[DirectMissingMethodProblem]("scala.collection.convert.JavaCollectionWrappers#JConcurrentMapWrapper.mapValuesInPlaceImpl"), // private[collection]
+
+    // #9733
+    ProblemFilters.exclude[MissingClassProblem]("scala.collection.concurrent.TrieMap$RemovalPolicy$"),                                        // private[concurrent]
+    // is this a MiMa bug? we really should need these two filters
+    //ProblemFilters.exclude[DirectMissingMethodProblem]("scala.collection.concurrent.TrieMap.removeRefEq"),                                    // private[concurrent]
+    //ProblemFilters.exclude[DirectMissingMethodProblem]("scala.collection.concurrent.TrieMap.replaceRefEq"),                                   // private[concurrent]
+    ProblemFilters.exclude[DirectMissingMethodProblem]("scala.collection.convert.JavaCollectionWrappers#JConcurrentMapWrapper.removeRefEq"),  // private[concurrent]
+    ProblemFilters.exclude[DirectMissingMethodProblem]("scala.collection.convert.JavaCollectionWrappers#JConcurrentMapWrapper.replaceRefEq"), // private[concurrent]
```

Comment:
I don't understand why we need MiMa filters for the `JConcurrentMapWrapper` ones, but not for the `TrieMap` ones. is this a bug?

Reply:
MiMa understands package private since verion 0.9.0, I assume that's why we don't need to add filters for the `private[concurrent]` methods `concurrent.Map.removeRefEq/replaceRefEq`.

`TrieMap.removeRefEq/replaceRefEq` are explicitly overridden (again with `private[concurrent]`), MiMa seems to understand that. `JConcurrentMapWrapper.removeRefEq/replaceRefEq` are compiler-generated mixin forwarders, which trip up MiMa.

If you look at the required filters for `filterInPlaceImpl/mapValuesInPlaceImpl`, both of these methods are defined in `concurrent.Map`, but need MiMa filters for the mixin forwarders in `TrieMap` and `JConcurrentMapWrapper`.

MiMa also doesn't catch that `TrieMap$RemovalPolicy$` is package private.

I'll create some tickets.